src/deadline/job_attachments/api/manifest.py

```diff
@@ -128,3 +173,78 @@ def _manifest_snapshot(
         # No manifest generated.
         logger.echo("No manifest generated")
         return None
+
+
+def _manifest_diff(
+    manifest: str,
+    root: str,
+    include: Optional[List[str]] = None,
+    exclude: Optional[List[str]] = None,
+    include_exclude_config: Optional[str] = None,
+    force_rehash=False,
+    logger: ClickLogger = ClickLogger(False),
+) -> ManifestDiff:
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to diff a manifest root with a previously snapshotted manifest.
+    :param manifest: Manifest file path to compare against.
+    :param root: Root directory to generate the manifest fileset.
+    :param include: Include glob to look for files to add to the manifest.
+    :param exclude: Exclude glob to exclude files from the manifest.
+    :param include_exclude_config: Config JSON or file containeing input and exclude config.
+    :param logger: Click Logger instance to print to CLI as test or JSON.
+    :returns: ManifestDiff object containing all new changed, deleted files.
+    """
+
+    # Find all files matching our regex
+    input_files = _glob_files(
+        root=root, include=include, exclude=exclude, include_exclude_config=include_exclude_config
+    )
+    input_paths = [Path(p) for p in input_files]
+
+    # Placeholder Asset Manager
+    asset_manager = S3AssetManager()
+
+    # parse the given manifest to compare against.
+    local_manifest_object: BaseAssetManifest
+    with open(manifest) as input_file:
+        manifest_data_str = input_file.read()
+        local_manifest_object = decode_manifest(manifest_data_str)
```

Comment:
Should we add file check and error handling here?

Reply:
`decode_manifest` raises errors, so if this failed, it will just fail the program outright. There's no need to do more error checks redundant.